pkg/auth/authenticator/password/ldappassword/util/server.go

```diff
@@ -0,0 +1,73 @@
+package util
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+
+	"github.com/go-ldap/ldap"
+)
+
+// LDAPServerConfig holds information for connecting to an LDAP server
+type LDAPServerConfig struct {
```

Comment:
> I'm not convinced that this is possible. Things LDAPServerConfig has that LDAPURL does not:

Default them and/or take params.  Probably different flavors for the different security options.